rust/lance/src/index/vector/ivf.rs

```diff
@@ -1511,4 +1511,50 @@ mod tests {
                 .all(|v| (0.0..2.0).contains(v)));
         }
     }
+
+    #[tokio::test]
+    async fn test_create_ivf_pq_dot() {
+        let test_dir = tempdir().unwrap();
+        let test_uri = test_dir.path().to_str().unwrap();
+
+        let (mut dataset, vector_array) = generate_test_dataset(test_uri).await;
+
+        let centroids = generate_random_array(2 * DIM);
+        let ivf_centroids = FixedSizeListArray::try_new_from_values(centroids, DIM as i32).unwrap();
+        let ivf_params = IvfBuildParams::try_with_centroids(2, Arc::new(ivf_centroids)).unwrap();
+
+        let codebook = Arc::new(generate_random_array(256 * DIM));
+        let pq_params = PQBuildParams::with_codebook(4, 8, codebook);
+
+        let params = VectorIndexParams::with_ivf_pq_params(MetricType::Dot, ivf_params, pq_params);
+
+        dataset
+            .create_index(&["vector"], IndexType::Vector, None, &params, false)
+            .await
+            .unwrap();
+
+        let sample_query = vector_array.value(10);
+        let query = sample_query.as_primitive::<Float32Type>();
+        let results = dataset
+            .scan()
+            .nearest("vector", query, 5)
+            .unwrap()
+            .try_into_stream()
+            .await
+            .unwrap()
+            .try_collect::<Vec<_>>()
+            .await
+            .unwrap();
+        assert_eq!(1, results.len());
+        assert_eq!(5, results[0].num_rows());
+
+        for batch in results.iter() {
+            let dist = &batch["_distance"];
+            assert!(dist
+                .as_primitive::<Float32Type>()
+                .values()
+                .iter()
+                .all(|v| (-2.0 * DIM as f32..0.0).contains(v)));
```

Comment:
Our dot product always returns negative value and with input value of `[0, 1)` we get a range of `(-1, 0]`. I suppose that is expected?